storage/innobase/xtrabackup/src/xbcloud/s3_ec2.h

```diff
@@ -0,0 +1,73 @@
+/******************************************************
+Copyright (c) 2022 Percona LLC and/or its affiliates.
+
+AWS EC2 Instance Metadata client implementation using IMDSv2.
+
+More details:
+https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html
+
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; version 2 of the License.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA
+
+*******************************************************/
+
+#ifndef XBCLOUD_S3_EC2_H
+#define XBCLOUD_S3_EC2_H
+
+#include "xbcloud/http.h"
+
+namespace xbcloud {
+class S3_ec2_instance {
+ private:
+  inline static const std::string host = "169.254.169.254";
```

Comment:
is it possible to not hardcode this IP? what if this stops working? if there is a domain name, it would be better.

Reply:
@satya-bodapati they currently do not have DNS for it (its a hardcoded link local IP that each instance has) and its implemented as per documentation:

https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html